microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -247,21 +247,26 @@ def verify_gpu_cuda_with_pytorch(
         _install_driver(node, log_path, log)
         _check_driver_installed(node, log)
 
-        _install_cudnn(node)
+        # pytorch and CUDA needs 4GB space to download and install
+        free_partition = node.find_partition_with_freespace(5)
 
-        gpu = node.features[Gpu]
+        _install_cudnn(node, log, free_partition)
 
         pip = node.tools[Pip]
         if not pip.exists_package("torch"):
-            pip.install_packages("torch")
+            pip.install_packages("torch", free_partition)
 
+        gpu = node.features[Gpu]
         gpu_script = "import torch;print(f'gpu count: {torch.cuda.device_count()}')"
         python = node.tools[Python]
         expected_count = gpu.get_gpu_count_with_lspci()
+        python_path = os.environ.get('PYTHONPATH', '')
 
+        torch_install_path = free_partition + "/python_packages"
```

Comment:
The `torch_install_path` is used once, so the variable not necessarily. And the path composition logic should be put together, so it's easy to read. See below,

```python
        python_path = os.environ.get('PYTHONPATH', '')
        python_path += f":{free_partition}/python_packages"

        script_result = python.run(
            f'-c "{gpu_script}"',
            force_run=True,
            update_envs={"PYTHONPATH": python_path}
        )
```